pennylane/tape/operation_recorder.py

```diff
@@ -63,20 +72,33 @@ def _process_queue(self):
         self.obs = new_tape.observables
 
     def __str__(self):
-        output = ""
-        output += "Operations\n"
-        output += "==========\n"
-        for op in self.ops:
-            output += repr(op) + "\n"
-
-        output += "\n"
-        output += "Observables\n"
-        output += "===========\n"
-        for op in self.obs:
-            output += repr(op) + "\n"
-
-        return output
+        return "\n".join(
+            [
+                "Operations",
+                "==========",
+                *[repr(op) for op in self.ops],
+                "",
+                "Observables",
+                "===========",
+                *[repr(op) for op in self.obs],
+                "",
+            ]
+        )
 
     @property
     def queue(self):
         return self.ops + self.obs
+
+    def __getitem__(self, key):
+        """
+        Overrides the default because OperationRecorder is both a QuantumScript and an AnnotatedQueue.
+
+        If key is an int, the caller is likely indexing the backing QuantumScript. Otherwise, the
+        caller is likely indexing the backing AnnotatedQueue.
+        """
+        if isinstance(key, int):
+            return QuantumScript.__getitem__(self, key)
+        return AnnotatedQueue.__getitem__(self, key)
+
+    def __setitem__(self, key, val):
+        AnnotatedQueue.__setitem__(self, key, val)

```

Comment:
Should we also add `__hash__` and return `QuantumScript.__hash__`? (like in `QuantumTape`)